src/gpu/intel/sdpa/micro.cpp

```diff
@@ -521,8 +521,8 @@ status_t micro_t::pd_t::init_conf(impl::engine_t *engine) {
         conf.prefetch_d_max = 0;
     }
 
-    const bool is_xe2 = pd->arch() == compute::gpu_arch_t::xe2;
-    conf.q_arrive_await_barrier = (Q > 1) && !is_xe2;
+    const bool arch_gte_xe2 = pd->arch() >= compute::gpu_arch_t::xe2;
```

Comment:
A comment would be nice explaining what tests fail so that we can come back to it.